src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -79,5 +81,17 @@ public static async Task ReadPrimitivesWithTrailingTriviaAsync()
                 Assert.Equal(1, i);
             }
         }
+
+        [Fact]
+        public static async Task ReadCollectionPassingNullValue()
+        {
+            using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes("null")))
+            {
+                JsonSerializerOptions options = new JsonSerializerOptions();
+
+                var collection  = await JsonSerializer.ReadAsync<IList<object>>(stream, options);
+                Assert.Null(collection);
```

Comment:
nit: Can you also add an attempt to read into a collection of a non-reference type in this test method, like `IList<int>`?

```C#
IList<int> anotherCollection = await JsonSerializer.ReadAsync<IList<int>>(stream);
Assert.Null(anotherCollection);
```